src/QMCHamiltonians/EwaldTools.h

```diff
@@ -0,0 +1,382 @@
+//////////////////////////////////////////////////////////////////////////////////////
+// This file is distributed under the University of Illinois/NCSA Open Source License.
+// See LICENSE file in top directory for details.
+//
+// Copyright (c) 2019 QMCPACK developers.
+//
+// File developed by: Jaron T. Krogel, krogeljt@ornl.gov, Oak Ridge National Laboratory
+//
+// File created by: Jaron T. Krogel, krogeljt@ornl.gov, Oak Ridge National Laboratory
+//////////////////////////////////////////////////////////////////////////////////////
+
+
+#ifndef EWALD_TOOLS_H
+#define EWALD_TOOLS_H
+
+#include "QMCHamiltonians/EwaldRef.h"
+
+namespace qmcplusplus
+{
+
+  using namespace ewaldref;
+
+
+
+template<typename T>
+real_t anisotropicGridSum(T& function, IntVec& nmax, bool zero = true, real_t tol = 1e-11)
+{
+  real_t dv  = std::numeric_limits<real_t>::max();
+  real_t dva = std::numeric_limits<real_t>::max();
+  real_t v   = 0.0;
+  int_t im   = 0;
+  int_t jm   = 0;
+  int_t km   = 0;
+  IntVec iv;
+  // Add the value at the origin, if requested.
+  if (zero)
```

Comment:
if we define a member function addZeroContribution() to both `T& function`, there is no need of the `zero` switch.

Reply:
The term should not need to know how it is being summed.

Reply:
ignore my last comment. zero and function need to be controlled independently.